app/views/sessions/purchase_cart.html.erb

```diff
@@ -7,7 +7,11 @@
     </tr>
     <% if session[:cart] %>
         <% session[:cart].each do |item_id, quantity| %>
-        <% item = Item.find_by(id: item_id) %>
+            <% item = Item.find_by(id: item_id) %>
+            <!-- Line below generates receipt for purchasing this item, if possible -->
+            <% if logged_in? %>
```

Comment:
Only returning a receipt if the user is logged in seems undesirable. If the user is not logged in the seller does not get a receipt. I suggest that the receipt be tied to some randomly generated confirmation number, that can then be used to look up the both the order and view the receipt, whether or not the user is logged in.